Mathlib/Data/Matrix/PEquiv.lean

```diff
@@ -55,13 +55,28 @@ theorem toMatrix_apply [DecidableEq n] [Zero α] [One α] (f : m ≃. n) (i j) :
     toMatrix f i j = if j ∈ f i then (1 : α) else 0 :=
   rfl
 
-theorem mul_matrix_apply [Fintype m] [DecidableEq m] [Semiring α] (f : l ≃. m) (M : Matrix m n α)
+theorem toMatrix_mul_apply [Fintype m] [DecidableEq m] [Semiring α] (f : l ≃. m) (M : Matrix m n α)
     (i j) : (f.toMatrix * M :) i j = Option.casesOn (f i) 0 fun fi => M fi j := by
   dsimp [toMatrix, Matrix.mul_apply]
   cases' h : f i with fi
   · simp [h]
   · rw [Finset.sum_eq_single fi] <;> simp +contextual [h, eq_comm]
 
+@[deprecated (since := "2025-01-27")] alias mul_matrix_apply := toMatrix_mul_apply
+
+theorem mul_toMatrix_apply [Fintype m] [Semiring α] [DecidableEq n] (M : Matrix l m α) (f : m ≃. n)
+    (i j) : (M * f.toMatrix :) i j = Option.casesOn (f.symm j) 0 (M i) := by
```

Comment:
Why smile? Is the following not what's meant?
```suggestion
    (i j) : ⇑(M * f.toMatrix) i j = Option.casesOn (f.symm j) 0 (M i) := by
```

Reply:
The smile was already here before this PR. Does your suggestion actually work?

Reply:
Hmm, indeed it doesn't